### DISCUSSION TITLE
I can't change the epoch

### DISCUSSION THREAD

**Discussion by Balporsugu21** [open-edge-platform/anomalib#1906] [Q&A] [answered]
# Import the required modules
from anomalib.data import MVTec
from anomalib.models import Patchcore
from anomalib.engine import Engine
from anomalib.models import Padim

# Initialize the datamodule, model and engine
datamodule = MVTec()
model = Padim()
engine = Engine(
   

    image_metrics=["AUROC"],
    accelerator="auto",
    check_val_every_n_epoch=1,
    limit_train_batches=2.0,
    limit_val_batches=2.0,
    limit_test_batches=2.0,
    precision=32, 
    log_every_n_steps=50,
    devices=1,
    max_epochs=3,
    num_sanity_val_steps=0,
    val_check_interval=1.0,
)

# Train the model
engine.fit(datamodule=datamodule, model=model)    This is my code but for some reason the epoch is only 1 please help me

**Comment by TurboJonte** [comment]
> Since the model does not require training, we limit the max_epochs to 1. Since we need to run training epoch before validation, we also set the sanity steps to 0

Source:
https://anomalib.readthedocs.io/en/latest/markdown/guides/reference/models/image/padim.html

**Reply by Balporsugu21** [reply]
I encounter the same problem when running  PatchCore and ganomaly. The epoch still does not change. What should I do? It automatically changes to 1000 epoch.

**Reply by samet-akcay** [reply, marked as answer]
You should not change Patchcore either. It always needs only 1 epoch to train the model. For ganomaly, we could check it out.

**Reply by Balporsugu21** [reply]
Thank you very much you helped a lot
